cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java

```diff
@@ -0,0 +1,170 @@
+     *
+     * @param credentialsProvider AWSCredentialsProvider instance.
+     */
+    CrossRegionReplicationMM2Converter(
+            GlueSchemaRegistryDeserializationFacade deserializationFacade,
+            GlueSchemaRegistrySerializationFacade serializationFacade,
+            AwsCredentialsProvider credentialsProvider,
+            GlueSchemaRegistryDeserializerImpl deserializerImpl,
+            GlueSchemaRegistrySerializerImpl serializerImpl) {
+
+        this.deserializationFacade = deserializationFacade;
+        this.serializationFacade = serializationFacade;
+        this.credentialsProvider = credentialsProvider;
+        this.deserializer = deserializerImpl;
+        this.serializer = serializerImpl;
+    }
+
+    /**
+     * Configure the MM2 Schema Replication Converter.
+     * @param configs configuration elements for the converter
+     * @param isKey true if key, false otherwise
+     */
+    @Override
+    public void configure(Map<String, ?> configs, boolean isKey) {
+        this.isKey = isKey;
+
+        credentialsProvider = DefaultCredentialsProvider.builder().build();
+
+        // Put the source and target regions into configurations respectively
+        Map<String, Object> sourceConfigs = new HashMap<>(configs);
+        Map<String, Object> targetConfigs = new HashMap<>(configs);
+
+
+        if (configs.get(AWSSchemaRegistryConstants.AWS_SOURCE_REGION) == null){
+            throw new DataException("Source Region is not provided.");
+        }
+        else if (configs.get(AWSSchemaRegistryConstants.AWS_TARGET_REGION) == null && configs.get(AWSSchemaRegistryConstants.AWS_REGION) == null){
+            throw new DataException("Target Region is not provided.");
+        }
+
+        sourceConfigs.put(AWSSchemaRegistryConstants.AWS_REGION, configs.get(AWSSchemaRegistryConstants.AWS_SOURCE_REGION));
+        targetConfigs.put(AWSSchemaRegistryConstants.AWS_REGION, configs.get(AWSSchemaRegistryConstants.AWS_TARGET_REGION));
+
+        deserializationFacade =
+                GlueSchemaRegistryDeserializationFacade.builder()
+                        .credentialProvider(credentialsProvider)
+                        .configs(sourceConfigs)
+                        .build();
+
+        serializationFacade   =
+                GlueSchemaRegistrySerializationFacade.builder()
+                        .credentialProvider(credentialsProvider)
+                        .configs(targetConfigs)
+                        .build();
+
+        serializer   = new GlueSchemaRegistrySerializerImpl(credentialsProvider, new GlueSchemaRegistryConfiguration(targetConfigs));
+
+        deserializer = new GlueSchemaRegistryDeserializerImpl(credentialsProvider, new GlueSchemaRegistryConfiguration(sourceConfigs));
+
+    }
+
+    @Override
+    public byte[] fromConnectData(String topic, org.apache.kafka.connect.data.Schema schema, Object value) {
+        byte[] bytes = (byte[]) value;
+        if (value == null) return new byte[0];
```

Comment:
If value is null, we cannot return empty bytes as empty bytes are different from null bytes. This can also be the first statement so that we don't risk processing nulls.

Reply:
Does it mean I should return null or null bytes (I was looking null bytes up but not so certain which one to 	use)